cmd/run/run.go

```diff
@@ -465,8 +465,9 @@ func (s *ServerContext) Run(ctx context.Context, config *serverconfig.Config) er
 		s.Logger.Info(fmt.Sprintf("📈 starting metrics server on '%s'", config.Metrics.Addr))
 
 		go func() {
-			http.Handle("/metrics", promhttp.Handler())
-			if err := http.ListenAndServe(config.Metrics.Addr, nil); err != nil {
+			mux := http.NewServeMux()
+			mux.Handle("/metrics", promhttp.Handler())
+			if err := http.ListenAndServe(config.Metrics.Addr, mux); err != nil {
```

Comment:
This change prevents

```
2023-10-27T11:07:41.557-0700	INFO	🚀 starting openfga service...	{"version": "dev", "date": "unknown", "commit": "none", "go-version": "go1.21.3"}
panic: http: multiple registrations for /metrics

goroutine 447 [running]:
net/http.(*ServeMux).Handle(0x31cac80, {0x25d3d41, 0x8}, {0x27fd440?, 0xc0005632f0})
	/usr/local/opt/go/libexec/src/net/http/server.go:2530 +0x3bd
net/http.Handle(...)
	/usr/local/opt/go/libexec/src/net/http/server.go:2573
github.com/openfga/openfga/cmd/run.(*ServerContext).Run.func2()
	/Users/mparnisari/GitHub/openfga/cmd/run/run.go:468 +0x88
created by github.com/openfga/openfga/cmd/run.(*ServerContext).Run in goroutine 440
	/Users/mparnisari/GitHub/openfga/cmd/run/run.go:467 +0x33c5
```

Reply:
Taking a look. There used to be a good prometheus test package to de-register between tests